src/Microsoft.TestPlatform.Client/DesignMode/DesignModeTestEventsRegistrar.cs

```diff
@@ -4,22 +4,21 @@
 namespace Microsoft.VisualStudio.TestPlatform.Client.DesignMode
 {
     using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
 
-    /// <summary>
-    /// Registers the discovery and test run events for design mode flow
-    /// </summary>
-    public class DesignModeTestEventsRegistrar : ITestDiscoveryEventsRegistrar, ITestRunEventsRegistrar
+    internal class IdentifiableDesignModeTestEventsRegistrar : ITestDiscoveryEventsRegistrar, ITestRunEventsRegistrar
     {
+        private string testRunId;
```

Comment:
nit: `readonly`, also the other one maybe

Reply:
Happy to enable the related rule as warning after we merge the `.editorconfig` change if we agree on it.